js/main-menu.js

```diff
@@ -132,8 +132,10 @@
    * @param {object} $menuItem
    */
   function wrapMenuItemLabelInHTML ($menuItem) {
-    $menuItem.contents().filter(function () {
-      return this.nodeType === 3 && $(this).parent().is($menuItem);
-    }).wrap('<span class="menumain-label" />')
+    var itemLabel = $('.crm-i, .fa', $menuItem)[0].nextSibling;
+
+    if (itemLabel.nodeType === Node.TEXT_NODE) {
```

Comment:
what if not? In the end we must have an element wrapped in `.menumain-label`, but from this code it seems that there is also the possibility that we might not have one

Reply:
I'm not sure I understand. If it's not a text note, then it will not be wrapped, just like the previous version. The only difference here is that the label node doesn't need to be a direct child of the menu item anymore